drivers/adc/adc_stm32.c

```diff
@@ -413,6 +413,68 @@ static int start_read(const struct device *dev,
 	LL_ADC_SetResolution(adc, resolution);
 #endif
 
+#if defined(CONFIG_SOC_SERIES_STM32G0X) || \
```

Comment:
I wonder if we would be able to define compatible that would allow to get rid of those ifdef..
But that's beyond the point of this PR